src/s3fs.cpp

```diff
@@ -59,6 +59,31 @@
 #define ENOATTR                   ENODATA
 #endif
 
+#define PRINT_FUSE_CTX(level, indent, fmt, ...) do {                \
+    struct fuse_context *ctx = fuse_get_context();                  \
```

Comment:
Could you change this condition about `if(S3fsLog::IsS3fsLogInfo()){...}` instead of `do{...}while(0);`?
This avoids unnecessary calls to `fuse_get_context()` when not at `INFO` level.

Reply:
Sorry, I didn't quite get it, fuse_get_context() is always necessary, no matter what the log level is. Would you please give more details?

Reply:
`S3FS_LOW_LOGPRN2` calling from `PRINT_FUSE_CTX` determines the log level in that macro (that is the `s3fs_low_logprn2` function) and works.
However, the `fuse_get_context` function that gets called is always executed every time `PRINT_FUSE_CTX` is called, regardless of log level.

If possible, I also wanted to avoid calling this function depending on the log level.
Therefore, I think it is better to judge the log level at the beginning of the `PRINT_FUSE_CTX` macro and not call the `fuse_get_context` function if it is not necessary.
It may not cost much, but I would like to avoid unnecessary calls (which end up calling `pthread` functions) if possible.

Reply:
I get it now, when current log level is not debug, s3fs_low_logprn2() won't try to print anything, so the fuse_get_context() call is wasted. Will update.